tools/modules/framework/interface_yes_nobox.sh

```
@@ -0,0 +1,42 @@
+	local allowed_functions=("process_input" "other_function")
+	# Check if the next_action is in the list of allowed functions
+	found=0
+	for func in "${allowed_functions[@]}"; do
+		if [[ "$func" == "$next_action" ]]; then
+			found=1
+			break
+		fi
+	done
+
+	if [[ "$found" -eq 1 ]]; then
+		if $($DIALOG --yesno "$message" 10 80 3>&1 1>&2 2>&3); then
```

Shellcheck (0.10.0)

[warning] Remove surrounding $() to avoid executing output (or use eval if intentional).

(SC2091)